docs/design/DATABASE_DESIGN.md

```
@@ -0,0 +1,412 @@
+- `WHERE user_id = current_user.id` だけで安全に絞り込める
+
+**テーブル分割しない理由**
+- 月ごとにテーブルを分けると前年同月比・トレンド分析が JOIN だらけになる
+- `effective_date` カラムにインデックスを貼れば絞り込みのパフォーマンスは十分
+- 個人利用レベルのデータ量ではパフォーマンス問題は起きない
+
+---
+
+## モデル構成
+
+```
```

markdownlint-cli2 (0.22.0)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)